pkg/schema/annotations.go

```
@@ -87,6 +93,43 @@ func NewNullableAnnotation(ann template.NodeAnnotation, valueType yamlmeta.Type,
 	return &NullableAnnotation{valueType, pos}, nil
 }
 
+// NewDefaultAnnotation checks the argument provided via @schema/default annotation, and returns wrapper for that value.
+func NewDefaultAnnotation(ann template.NodeAnnotation, inferredType yamlmeta.Type, pos *filepos.Position) (*DefaultAnnotation, error) {
+	if len(ann.Kwargs) != 0 {
+		return nil, schemaAssertionError{
+			position:    pos,
+			description: fmt.Sprintf("expected @%v annotation to contain value with type of annotated node", AnnotationDefault),
+			expected:    fmt.Sprintf("%s (by %s)", inferredType.String(), inferredType.GetDefinitionPosition().AsCompactString()),
+			found:       fmt.Sprintf("%T", ann.Kwargs[0]),
+			hints: []string{
+				"value must be the same type as the annotated node",
+				"value must be in Starlark format, e.g.: {'key': 'value'}, True"},
+		}
+	}
+	if len(ann.Args) == 0 {
+		return nil, schemaAssertionError{
+			position:    pos,
+			description: fmt.Sprintf("expected @%v annotation to contain default value", AnnotationDefault),
+			expected:    fmt.Sprintf("%s (by %s)", inferredType.String(), inferredType.GetDefinitionPosition().AsCompactString()),
+			found:       "missing value",
+			hints: []string{
+				"value must be the same type as the annotated node",
+				"value must be in Starlark format, e.g.: {'key': 'value'}, True"},
+		}
+	}
+	val, err := core.NewStarlarkValue(ann.Args[0]).AsGoValue()
```

To call this feature "done", we need to support _any_ Starlark expression (e.g. a function call)
Do that in _this_ story? or write another story?

---

talking about this synchronously...

In fact, by the time this annotation is processed, all of the Starlark has been evaluated. Therefore the only other type are function types (i.e. lambdas). Those are already not supported in Data Values.

---

Instead of leaving this untested error in the code, we want to panic to signal that something has unexpectedly gone wrong.